### PR TITLE
[#104164510] Adds screenshot tutorial

### PR DESCRIPTION
**Pull request by Kuranes**
uses glReadPixel, freeing the need for preserveDrawingBuffer performance impacts.

### CODE REVIEW

**Comment by Kuranes** [comment]
Hitting a wall on #399
because the glTrace show that we apply a default "gl.viewport(800,600)" breaking glReadPixel read.

**Comment by Kuranes** [comment]
it' just about adding a camera lastDraw callback here:
https://github.com/cedricpinson/osgjs/blob/develop/sources/osg/RenderStage.js#L129
as done in OSG, see here
https://github.com/openscenegraph/osg/blob/f1a0ff43c6bd47387a480de89f35fff2d0ed7a95/src/osgUtil/RenderStage.cpp#L1316

**Comment by cedricpinson** [comment]
@Kuranes yes you can implement it using the same method as in osg

File: sources/osgViewer/Viewer.js [hunk 1]
```diff
@@ -449,7 +449,9 @@ define( [
         beginFrame: function () {
             this._startFrameTick = Timer.instance().tick();
         },
-
+        setEndFrameCallBack: function ( u ) {
```
Comment (cedricpinson) [inline sources/osgViewer/Viewer.js]: why this accerssors ? is viewer.getCamera().setFinalDrawCallback( cb ) not enough ?

Comment (Kuranes) [inline sources/osgViewer/Viewer.js]: mm first iteration added that before the camera callback. 
Not needed here, still would find that useful in many case/experiments with code.
can remove

**Pull request closed without merging**